2-mandatory/3-financial-times.js

```diff
@@ -4,36 +4,80 @@
     The home page of the web site has a headline section, which only has space for article titles which are 65 characters or less.
     Implement the function below, which will return a new array containing only article titles which will fit.
 */
+
 function potentialHeadlines(allArticleTitles) {
-    // TODO
+    return allArticleTitles.filter(element => element.length <= 65)
 }
 
+// console.log(potentialHeadlines(ARTICLE_TITLES));
+// ["British companies look to muscle in on US retail investing boom", "Libor to take firm step towards oblivion on New Year's Day", "Audit profession unattractive to new recruits, says PwC boss", "The three questions that dominate investment"]
+
 /*
     The editor of the FT likes short headlines with only a few words!
     Implement the function below, which returns the title with the fewest words.
     (you can assume words will always be seperated by a space)
 */
+
 function titleWithFewestWords(allArticleTitles) {
-    // TODO
+    return allArticleTitles.sort((elementOne, elementTwo) => elementOne.split(" ").length - elementTwo.split(" ").length)[0]
+
+    // return allArticleTitles[allArticleTitles.map((element) => element.split(" ").length).indexOf(Math.min(...allArticleTitles.map((element) => element.split(" ").length)))];
+    // ^ this is ridiculous but it works allArticleTitles[dynamic index]
+
+    // allArticleTitles.reduce((pV,cV) => pV.split(" ").length <= cV.split(" ").length ? pV : cV);
+    // ^ I WANT TO LEARN HOW TO DO THIS WITH REDUCE -- HELP!
 }
 
+// console.log(titleWithFewestWords(ARTICLE_TITLES));
+// "The three questions that dominate investment"
+
+
 /*
     The editor of the FT has realised that headlines which have numbers in them get more clicks!
     Implement the function below to return a new array containing all the headlines which contain a number.
     (Hint: remember that you can also loop through the characters of a string if you need to)
 */
+
 function headlinesWithNumbers(allArticleTitles) {
-    // TODO
+    // const numbers = ["0","1","2","3","4","5","6","7","8","9"];
+    // let result = [];
+    // for (let i = 0; i < allArticleTitles.length; i++) {
+    //     for (let j = 0; j < allArticleTitles[i].length; j++) {
+    //         if (numbers.includes(allArticleTitles[i][j])) {
+    //             result.push(allArticleTitles[i]);
+    //             break;
+    //         }
+    //     }
+    // }
+    // return result;
+
+    // return allArticleTitles.filter(element => /\d+/.test(element));
+    // ^ i refuse to use regex :)
+
+    return allArticleTitles.filter(element => [...element].find(number => number >= "0" && number <= "9"));
```

Comment:
This works fine, but maybe worth considering a regex instead, it would read a bit more simply?
```
allArticleTitles.filter(title => /\d/.test(title))
```